src/elliptic/curves/ed25519.rs

```diff
@@ -504,23 +506,24 @@ impl<'de> Visitor<'de> for RistrettoCurvPointVisitor {
     type Value = Ed25519Point;
 
     fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
-        formatter.write_str("ed25519CurvPoint")
+        formatter.write_str("Ed25519Point")
     }
 
     fn visit_map<E: MapAccess<'de>>(self, mut map: E) -> Result<Ed25519Point, E::Error> {
-        let mut bytes_str: String = "".to_string();
-
-        while let Some(key) = map.next_key::<&'de str>()? {
-            let v = map.next_value::<&'de str>()?;
-            match key {
-                "bytes_str" => {
-                    bytes_str = String::from(v);
-                }
-                _ => panic!("deSerialization failed!"),
+        let mut bytes_str: String = String::new();
+
+        while let Some(ref key) = map.next_key::<String>()? {
+            let v = map.next_value::<String>()?;
+            if key == "bytes_str" {
```

Comment:
why this change is required?

Reply:
I pushed a revert to this change. Probably just was something needed for the work in progress.